MAPREDUCE-7497. mapreduce tests have stopped running.

slfan1989 opened the pull request:
<!--
  Thanks for sending a pull request!
    1. If this is your first time, please read our contributor guidelines: https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute
    2. Make sure your PR title starts with JIRA issue id, e.g., 'HADOOP-17799. Your PR title ...'.
-->

### Description of PR

JIRA: MAPREDUCE-7497. mapreduce tests have stopped running.

After completing [HADOOP-15984](https://issues.apache.org/jira/browse/HADOOP-15984), we added JUnit5 dependencies to some modules. These dependencies caused maven-surefire to fail to recognize JUnit4 tests, resulting in the MapReduce module's unit tests stopping. We will track and fix this issue in this JIRA.

### How was this patch tested?

Unit Test.

### For code changes:

- [X] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [x] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

slfan1989 committed:
MAPREDUCE-7497. mapreduce tests have stopped running.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 50s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  38m 39s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 54s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  17m 45s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 40s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   6m 21s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m 55s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   5m 27s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +0 :ok: |  spotbugs  |   0m 41s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  39m 23s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 38s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 19s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  17m 19s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 35s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   6m 11s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   5m 49s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   5m 32s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +0 :ok: |  spotbugs  |   0m 37s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  38m 58s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 34s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 12s |  |  hadoop-yarn-api in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 51s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 57s |  |  hadoop-mapreduce-client-shuffle in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 54s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 22s |  |  hadoop-mapreduce-client-hs in the patch passed.  |
| +1 :green_heart: |  unit  | 126m 27s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  unit  |  11m 12s |  |  hadoop-mapreduce-client-nativetask in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  9s |  |  The patch does not generate ASF License warnings.  |
|  |   | 435m 14s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7343 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle |
| uname | Linux 93832fadda53 5.15.0-125-generic #135-Ubuntu SMP Fri Sep 27 13:53:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 2e3f5e67fafead4686b376b308505b85c3daf533 |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/1/testReport/ |
| Max. process+thread count | 1160 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-nativetask U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

cnauroth commented on hadoop-project/pom.xml:
```diff
@@ -185,8 +185,8 @@
     <!-- Plugin versions and config -->
-    <maven-surefire-plugin.argLine>-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError ${extraJavaTestArgs}</maven-surefire-plugin.argLine>
-    <maven-surefire-plugin.version>3.0.0-M1</maven-surefire-plugin.version>
+    <maven-surefire-plugin.argLine>-Xmx2048m -Xss2m -XX:+HeapDumpOnOutOfMemoryError ${extraJavaTestArgs}</maven-surefire-plugin.argLine>
```
Does the default stack size not work?

slfan1989 replied:
This change was made because the `ResourceManager` module in #7335 reported an `unable to create new native thread` exception. Additionally, the third build result of #7335 appeared abnormal, and I suspected that the issue might be caused by the large number of compiled modules. I increased the `-Xss` size, which seemed to have some effect. The default value for this parameter is 1m.

The build report is as follows:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7335/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt

If either #7343 or #7335 is merged, the other PR can remove this change.

cnauroth commented on hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/api/records/timelineservice/ApplicationEntity.java:
```diff
@@ -43,7 +43,12 @@ public ApplicationEntity(TimelineEntity entity) {
 
   public String getQueue() {
-    return getInfo().get(QUEUE_INFO_KEY).toString();
+    if (getInfo() != null) {
```
Is this an unrelated change?

slfan1989 replied:
This is a necessary change, and I have provided a detailed explanation of the modification in the comment below.

cnauroth replied:
Thank you for the explanation.

Is this potentially a backward-incompatible change? For example, does it mean clients may now be getting back a queue element with an empty string, where previously it was omitted entirely from the response?

slfan1989 replied:
Thank you very much for your comment! I have further refined this part of the code and added the `@JsonInclude(JsonInclude.Include.NON_EMPTY)` annotation to ensure compatibility of this logic.

slfan1989 replied:
From my perspective, we need to continue iterating and improving the YARN Timeline-related code, implementing parsing for both JSON and XML formats in the way Jersey2 does. I hope that after completing the JDK17 support, we can continue advancing these improvements.

slfan1989 commented:
@cnauroth Thank you very much for reviewing this PR! I apologize for not providing a detailed explanation of why we made these changes. Let me explain in detail why we made these modifications.

After upgrading Jersey to version 2.46, we found that Jersey 2 handles requests differently compared to Jersey 1. Overall, it has become stricter.

- Change 1: TestHsWebServices#testInvalidAccept

This unit test is for the following interface: AMWebServices#get, with the code as follows:

https://github.com/apache/hadoop/blob/44a5cba78ac7d29a72d3452fbd807f31ae90c325/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/main/java/org/apache/hadoop/mapreduce/v2/app/webapp/AMWebServices.java#L239-L245

This interface only accepts `MediaType.APPLICATION_JSON` and `MediaType.APPLICATION_XML` types, and does not accept `MediaType.TEXT_PLAIN`. Therefore, if we call it with `MediaType.TEXT_PLAIN`, a `NotAcceptableException` will be thrown.

Therefore, we need to improve this unit test.

- Change 2: TestHsWebServices#testInvalidAccept

TestHsWebServices#testInvalidAccept also requires improvement for reasons similar to Change 1.

- Change 3: ApplicationEntity#getQueue

ApplicationEntity#getQueue is a necessary change. We made this modification because the unit test was failing previously.

We can see this error at the following link:

[patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-jobclient.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7335/3/artifact/out/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-jobclient.txt)

```
[ERROR] org.apache.hadoop.mapred.TestMRTimelineEventHandling.testMRNewTimelineServiceEventHandling  Time elapsed: 87.424 s  <<< FAILURE!
java.lang.AssertionError: Expected event : JOB_FINISHED not found in the file /home/jenkins/jenkins-home/workspace/hadoop-multibranch_PR-7335/ubuntu-focal/src/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/target/TestMRTimelineEventHandling-test_dir/timeline_service_data/entities/yarn_cluster/jenkins/test-job-succeed/1/1738269699937/application_1738269698564_0001/MAPREDUCE_JOB/job_1738269698564_0001.thist
	at org.junit.Assert.fail(Assert.java:89)
.......
```

This is a subtle error, and it took me a long time to identify the issue. The problem arises from the difference in how Jersey 1 and Jersey 2 handle complex type conversions to JSON. In Jersey 1, we could customize the conversion process, but in Jersey 2, we need to define custom JSON readers and writers. In my case, I defined a `TimelineEntitiesWriter.java`, which relies on `com.fasterxml.jackson.databind.ObjectMapper` for the conversion. If the object contains null values, a `NullPointerException` occurs during the conversion.

Therefore, I have implemented handling for the conversion of this field.

Since it is the timeclient-v2 version, it will eventually reach the `putEntities` method in `TimelineCollectorWebService`.

https://github.com/apache/hadoop/blob/44a5cba78ac7d29a72d3452fbd807f31ae90c325/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice/src/main/java/org/apache/hadoop/yarn/server/timelineservice/collector/TimelineCollectorWebService.java#L135-L167

Inside, there is a complex data type, `TimelineEntities`, and its parsing relies on the `TimelineEntitiesWriter.java` class that I wrote.

https://github.com/apache/hadoop/blob/44a5cba78ac7d29a72d3452fbd807f31ae90c325/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/api/records/timelineservice/writer/TimelineEntitiesWriter.java#L40-L65

This will throw a `NullPointerException` because the queue is null. Therefore, I implemented error handling in the `getQueue` method to ensure that it does not throw a `NullPointerException`.

Commit:
YARN-11764. Add the JsonInclude annotation.

cnauroth reviewed (approved):
+1, pending completion of the pre-submit run on the latest changes.

Thank you, @slfan1989 !

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  18m 22s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  8s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  36m  1s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 41s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  18m 53s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 47s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   6m 12s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m 48s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   5m 24s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +0 :ok: |  spotbugs  |   0m 41s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  38m 49s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 12s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 46s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 57s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  17m 57s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 46s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   6m  9s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   5m 40s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   5m 27s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +0 :ok: |  spotbugs  |   0m 36s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  38m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 34s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 12s |  |  hadoop-yarn-api in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 53s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 57s |  |  hadoop-mapreduce-client-shuffle in the patch passed.  |
| +1 :green_heart: |  unit  |   9m 33s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 21s |  |  hadoop-mapreduce-client-hs in the patch passed.  |
| +1 :green_heart: |  unit  | 125m 29s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  unit  |  11m 12s |  |  hadoop-mapreduce-client-nativetask in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  |  The patch does not generate ASF License warnings.  |
|  |   | 449m 38s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7343 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle |
| uname | Linux 2d6722bbf7e8 5.15.0-125-generic #135-Ubuntu SMP Fri Sep 27 13:53:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a5b718ebdbd843271735f7037bf7a180d8ce6e13 |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/2/testReport/ |
| Max. process+thread count | 1161 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-nativetask U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 committed:
Merge branch 'apache:trunk' into MAPREDUCE-7497

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 51s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 26s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  36m 13s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  19m 29s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  17m 32s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  checkstyle  |   4m 42s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 38s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m 13s |  |  trunk passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 47s |  |  trunk passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   8m 28s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  39m  7s |  |  branch has no errors when building and testing our client artifacts.  |
| -0 :warning: |  patch  |  39m 36s |  |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 45s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  18m 45s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 30s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  javac  |  17m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 32s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   5m 36s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   5m 10s |  |  the patch passed with JDK Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   4m 49s |  |  the patch passed with JDK Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga  |
| +1 :green_heart: |  spotbugs  |   9m 48s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  38m 54s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 13s |  |  hadoop-yarn-api in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 48s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 57s |  |  hadoop-mapreduce-client-shuffle in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 48s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 21s |  |  hadoop-mapreduce-client-hs in the patch passed.  |
| +1 :green_heart: |  unit  | 126m 38s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  unit  |  11m 12s |  |  hadoop-mapreduce-client-nativetask in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  |  The patch does not generate ASF License warnings.  |
|  |   | 425m 59s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7343 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle |
| uname | Linux 2802ed0f7e5f 5.15.0-125-generic #135-Ubuntu SMP Fri Sep 27 13:53:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 4cc7418279c29cf8e0cd7e1b8d16bebf0ad8e727 |
| Default Java | Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.25+9-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_432-8u432-ga~us1-0ubuntu2~20.04-ga |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/3/testReport/ |
| Max. process+thread count | 1135 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-nativetask U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7343/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

slfan1989 commented:
@cnauroth Thank you very much for reviewing the code! I will merge this PR into the trunk branch to begin the upgrade of the MapReduce module from `JUnit4` to `JUnit5`. Yetus feedback indicates that this PR meets the merge requirements.
